#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_scim_x11_frontend-ic-focus-LTC27940-215953.dpatch by Ikuya Awashiro <ikuya@fruitsbasket.info>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: This patch should enable input in Java swing apps.  It originated from the Fedora Project.
## DP: Fedora stopped applying the patch a while ago, saying it wasn't needed anymore.  User comments
## DP: indicate this is not the case.  See LP #178742.

@DPATCH@
diff -urNad scim-1.4.7~/modules/FrontEnd/scim_x11_frontend.cpp scim-1.4.7/modules/FrontEnd/scim_x11_frontend.cpp
--- scim-1.4.7~/modules/FrontEnd/scim_x11_frontend.cpp	2007-06-26 14:31:52.000000000 +0000
+++ scim-1.4.7/modules/FrontEnd/scim_x11_frontend.cpp	2008-03-15 06:36:15.000000000 +0000
@@ -1086,7 +1086,8 @@
     // If the ic is not focused, then return.
     if (!is_focused_ic (ic)) {
         SCIM_DEBUG_FRONTEND(1) << "IC " << call_data->icid << " is not focused, focus it first.\n";
-        ims_set_ic_focus_handler (ims, (IMChangeFocusStruct *) call_data);
+        // ims_set_ic_focus_handler (ims, (IMChangeFocusStruct *) call_data);
+        return 1;
     }
 
     XKeyEvent *event = (XKeyEvent*) &(call_data->event);
